runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -370,6 +370,12 @@ TR_J9ServerVM::jitFieldsAreSame(TR_ResolvedMethod * method1, I_32 cpIndex1, TR_R
          UDATA field1 = 0, field2 = 0;
          J9Class *declaringClass1 = NULL, *declaringClass2 = NULL;
 
+         // do a quick check to see if a field is unresolved for the current compilation
+         auto compInfoPT = static_cast<TR::CompilationInfoPerThreadRemote *>(_compInfoPT);
+         if (compInfoPT->getCachedUnresolvedField(ramClass1, cpIndex1) ||
+             compInfoPT->getCachedUnresolvedField(ramClass2, cpIndex2))
+            return false;
```

Comment:
This early return circumvents the mechanism that verifies the integrity of our caches. 
In debug mode we need to issue the remote call anyway and verify that the outcome was the same.

Reply:
But this per-compilation cache might contain outdated data, so outcome wouldn't be the same.

Reply:
ok. agreed